docs/internals/layout_in_storage.rst

```diff
@@ -128,7 +128,7 @@ The type of the value is ``uint256``, so it uses a single slot.
 ------------------------
 
 ``bytes`` and ``string`` are encoded identically.
-In general, the encoding is similar to ``bytes1[]``, in the sense that there is a slot for the array itself and
+In general, the encoding is similar to ``bytes1[]``, in the sense that there is a slot for the array length itself and
 a data area that is computed using a ``keccak256`` hash of that slot's position.
 However, for short values (shorter than 32 bytes) the array elements are stored together with the length in the same slot.
```

Comment:
I agree with @chriseth that the change does not necessarily clarify things because the slot is not meant only for length.

If we were to expand this, this would be my proposal:
```
In general, the encoding is similar to ``bytes1[]``, in the sense that it consists of one fixed slot and
a separately allocated data area whose location is computed using a ``keccak256`` hash of that slot's position.
However, the fixed slot is not used exclusively to store the length.
For short values (shorter than 32 bytes) the array elements are stored together with the length in
the fixed slot and the layout can dynamically change when elements are added or removed.
```